code/modules/surgery/organs/blood.dm

```diff
@@ -231,15 +231,26 @@
 
 	var/datum/reagent/blood/injected = get_blood(container.reagents)
 
+	if(!istype(injected)) //Sanity..
+		return
+
 	if(species && species.flags & NO_BLOOD)
 		reagents.add_reagent("blood", amount, injected.data)
 		reagents.update_total()
 		return
 
 	var/datum/reagent/blood/our = get_blood(vessel)
 
-	if(!istype(injected) || !istype(our))
+	if(!istype(our)) // Do we not have blood?
+		vessel.add_reagent("blood", amount, injected.data) // This is only ran once, so we need to add back thier DNA.
+		vessel.update_total()
+		..()
+		our = get_blood(vessel) // Get the new blood.
+		our.data["blood_DNA"] = src.dna.unique_enzymes // Give it the proper DNA we had before
+		our.data["blood_type"] = src.dna.b_type  // Proper blood type.
```

Comment:
Yeah, I'm not used to Byond and just having dna.b_type bothered me a bit too much.

I actually did do my testing with other mobs before hand, and then asked for a code review on BS12's #codershuttle channel and they said it looked fine so I made the PR. I (or a maintainer) can change it if it matters that much.